### PR TITLE
[ci] Migrate checks to GHA

### PR DESCRIPTION
**Pull request by alvicsam**
PR migrates jobs `quick-benchmarks`, `cargo-clippy`, `check-try-runtime` and `check-core-crypto-features` from Gitlab to GitHub

cc https://github.com/paritytech/ci_cd/issues/1006

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -88,3 +97,11 @@ jobs:
...
           ./check-features-variants.sh
           cd -
+
+  confirm-required-jobs-passed:
+    runs-on: ubuntu-latest
+    name: All checks passed
+    # If any new job gets added, be sure to add it to this array
+    needs: [cargo-clippy, check-try-runtime, check-core-crypto-features]
+    steps:
+      - run: echo '### Good job! All the tests passed 🚀' >> $GITHUB_STEP_SUMMARY
```
Comment (mordamax) [inline .github/workflows/checks.yml:107]:
```suggestion
      - run: echo '### Good job! All the checks passed 🚀' >> $GITHUB_STEP_SUMMARY
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -63,21 +75,21 @@ jobs:
...
           ./list-syscalls.rb ../../../target/x86_64-unknown-linux-musl/production/polkadot-prepare-worker --only-used-syscalls | diff -u prepare-worker-syscalls -
-    # todo:
-    # after_script:
-    # - if [[ "$CI_JOB_STATUS" == "failed" ]]; then
-    #   printf "The x86_64 syscalls used by the worker binaries have changed. Please review if this is expected and update polkadot/scripts/list-syscalls/*-worker-syscalls as needed.\n";
-    #   fi
+      - name: on_failure
+        if: failure() && steps.test.outcome == 'failure'
+        run: |
+          echo "The x86_64 syscalls used by the worker binaries have changed. Please review if this is expected and update polkadot/scripts/list-syscalls/*-worker-syscalls as needed." >> $GITHUB_STEP_SUMMARY
```
Comment (mordamax) [inline .github/workflows/tests.yml:88]: I did it in https://github.com/paritytech/polkadot-sdk/pull/5385 so could lead to conflicts

**Review by AndWeHaveAPlan — APPROVED** [review]

**Review by mordamax — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (alvicsam) [reply .github/workflows/tests.yml:89]: I checked your version and liked my implementation more 😛 but open for discussion

Comment (mordamax) [reply .github/workflows/tests.yml:89]: then unrequire the tests, merge my, and merge conflicts here with what you like more 😅😅

**Pull request merged**